Trouble building UCX (error utilizing binutils)

I am trying to build UCX from source using gcc7 + binutils 2.27. Whenever I run `make`, I get the following:

In file included from debug/debug.c:31:0:
/apps/default/include/bfd.h:6588:3: error: expected specifier-qualifier-list before 'ENUM_BITFIELD'
   ENUM_BITFIELD (bfd_format) format : 3;
   ^~~~~~~~~~~~~
/apps/default/include/bfd.h: In function 'bfd_set_cacheable':
/apps/default/include/bfd.h:6860:7: error: 'bfd {aka struct bfd}' has no member named 'cacheable'
   abfd->cacheable = val;
       ^~
debug/debug.c: In function 'load_file':
/apps/default/include/bfd.h:518:41: error: 'bfd {aka struct bfd}' has no member named 'flags'
 #define bfd_get_file_flags(abfd) ((abfd)->flags)
                                         ^
debug/debug.c:200:10: note: in expansion of macro 'bfd_get_file_flags'
     if ((bfd_get_file_flags(file->abfd) & HAS_SYMS) == 0) {
          ^~~~~~~~~~~~~~~~~~
make[2]: *** [debug/libucs_la-debug.lo] Error 1
make[2]: Leaving directory `/scratch/build/ucx-1.4.0/src/ucs'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/scratch/build/ucx-1.4.0'
make: *** [all] Error 2

Is there a particular version of binutils that must be used?

--- a/config/m4/ucs.m4
+++ b/config/m4/ucs.m4
@@ -37,13 +37,7 @@ AS_IF([test "x$enable_backtrace_detail" == xyes],
 	[
 	BT=1
 	AC_CHECK_HEADER([bfd.h], [], [AC_MSG_WARN([binutils headers not found])]; BT=0)
-	AC_CHECK_HEADERS([libiberty.h libiberty/libiberty.h], [], [],
-					[#define HAVE_DECL_BASENAME 1])
-	if test "x$ac_cv_header_libiberty_h" == "x" && test "x$ac_cv_header_libiberty_libiberty_h" == "x"; then
-	    AC_MSG_WARN([binutils headers not found]); BT=0
-	fi
 	AC_CHECK_LIB(bfd, bfd_openr,  LIBS="$LIBS -lbfd", [AC_MSG_WARN([bfd library not found])];BT=0)
-	AC_CHECK_LIB(iberty, xstrerror, LIBS="$LIBS -liberty", [AC_MSG_WARN([iberty library not found])];BT=0)
 	AC_CHECK_LIB(dl, dlopen, LIBS="$LIBS -ldl", [AC_MSG_WARN([dl library not found])];BT=0)
 	AC_CHECK_LIB(intl, main, LIBS="$LIBS -lintl", [AC_MSG_WARN([intl library not found])])
 	AC_CHECK_TYPES([struct dl_phdr_info], [], [AC_MSG_WARN([struct dl_phdr_info not defined])];BT=0,

--- a/src/ucs/debug/debug.c
+++ b/src/ucs/debug/debug.c
@@ -23,11 +23,6 @@
 #include <link.h>
 #include <dirent.h>
 #ifdef HAVE_DETAILED_BACKTRACE
-#  if HAVE_LIBIBERTY_H
-#    include <libiberty.h>
-#elif HAVE_LIBIBERTY_LIBIBERTY_H
-#    include <libiberty/libiberty.h>
-#  endif
 #  include <bfd.h>
 #endif /* HAVE_DETAILED_BACKTRACE */
 

